Incident Summary AI Agent (#2584)
port-labs/port-docs: docs/guides/all/generate-incident-updates-with-ai.md

---
displayed_sidebar: null
description: Learn how to leverage Port's AI capabilities to enhance incident management with contextual incident updates and automated Slack notifications.
---

# Generate incident updates with AI

> This guide demonstrates how to enhance your incident management workflow using Port's AI capabilities. You will learn how to create an AI agent that generates incident updates and sends slack notification with relevant technical context.

Reviewer: ```suggestion
When you’re busy fixing an incident, you don’t have time to write updates. Yes, others still need to know what’s going on. In this guide, you will learn to build an AI agent that generates incident updates and sends Slack notifications with relevant technical context.
```

- [GitHub integration](https://docs.port.io/build-your-software-catalog/sync-data-to-catalog/git/github/) is installed in your account.
- You have access to [create and configure AI agents](https://docs.port.io/ai-agents/overview#getting-started-with-ai-agents) in Port.

:::tip Alternative integrations
While this guide uses PagerDuty and GitHub, you can adapt it for other incident management tools like OpsGenie or FireHydrant, and other Git providers like GitLab or Azure DevOps.
:::

## Set up data model

We will create and configure blueprints to support our AI-enhanced incident management workflow. This includes setting up the incident data model and enriching it with repository metadata.

### Update the incident and service blueprints

To enable AI-generated incident updates and Slack notifications, update your blueprints as follows:

1. **Add an `ai_update` property to the incident blueprint**
   - Go to the [builder](https://app.getport.io/settings/data-model) page of your portal.
   - Find and select your incident blueprint (e.g., `pagerdutyIncident`).
   - Click on `{...} Edit JSON`.
   - Add the following property to the `properties` section:
    
      <details>

>       <summary><b>AI update property (Click to expand)</b></summary>

Reviewer 2: ```suggestion
      <summary><b>AI update property (click to expand)</b></summary>
```

   - Add the following property to the `properties` section:
    
      <details>
      <summary><b>Slack channel property (Click to expand)</b></summary>

      ```json showLineNumbers
      "slack_channel": {
        "type": "string",
        "title": "Slack Channel"
      }
      ```
      </details>

   - Click `Save` to update the blueprint.

      :::tip Slack channel configuration
      This step is optional. You can choose to:
      - Send notifications to a static channel (hardcoded in the automation)
      - Store the Slack channel in other hierarchies (e.g., on the incident itself)
      - Use different channels based on incident severity or team
      :::

### Update repository blueprint with additional context

When installing the GitHub app, the `Repository` blueprint is created by default. However, we need to add additional properties to enrich incident updates with repository metadata.

1. Go to the [builder](https://app.getport.io/settings/data-model) page of your portal.
2. Find and select your existing repository blueprint (e.g., `githubRepository`).
3. Click on `{...} Edit JSON`.
4. Add the following properties to the `properties` section:

    <details>

>     <summary><b>Additional repository properties (Click to expand)</b></summary>

Reviewer 2: ```suggestion
    <summary><b>Additional repository properties (click to expand)</b></summary>
```

    ```json showLineNumbers
    "last_contributor": {
      "title": "Last contributor",
      "icon": "TwoUsers",
      "type": "string",
      "format": "user"
    },
    "last_push": {
      "icon": "GitPullRequest",
      "title": "Last push",
      "description": "Last commit to the main branch",
      "type": "string",
      "format": "date-time"
    },
    "vulnerabilities": {
      "type": "number",
      "title": "Known Vulnerabilities",

>       "description": "Number of critical security vulnerabilities",

Reviewer 2: ```suggestion
      "description": "Number of critical security vulnerabilities"
```

      "trigger": {
        "type": "self-service",
        "operation": "DAY-2",
        "userInputs": {
          "properties": {
            "ai_update": {
              "title": "AI Update",
              "type": "string",
              "format": "markdown"
            }
          },
          "required": [
            "ai_update"
          ],
          "order": [
            "ai_update"
          ]
        },
        "blueprintIdentifier": "pagerdutyIncident"
      },
      "invocationMethod": {
        "type": "WEBHOOK",
        "url": "https://api.getport.io/v1/blueprints/pagerdutyIncident/entities/{{ .entity.identifier }}",
        "agent": false,
        "synchronized": true,
        "method": "PATCH",
        "headers": {
          "RUN_ID": "{{ .run.id }}",
          "Content-Type": "application/json"
        },
        "body": {
          "properties": {
            "ai_update": "{{ .inputs.ai_update }}"
          }
        }
      },
      "requiredApproval": false
    }
    ```
    </details>

5. Click `Save` to create the action.

## Create AI agent

Next, we will create an AI agent that generates helpful incident updates with contextual information.

> ### Configure the incident summary AI agent

Reviewer: ```suggestion
### Configure the incident update AI agent
```
